src/components/activity-card/activity-card.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+  ActivityBody,
+  RecentEventsBody,
+} from '@console/dynamic-plugin-sdk/provisional';
+import {
+  useK8sWatchResource,
+} from "@console/dynamic-plugin-sdk/api";
+import { FirehoseResource } from "@console/dynamic-plugin-sdk";
+import {getNamespace} from "../../selectors/index";
+//import { IBM_STORAGE_ODF_OPERATOR } from '../../constants/index';
+import {
+   PersistentVolumeClaimModel, 
+   EventModel,    
+  } from '../../models';
+import './activity-card.scss';
+import {GetFlashSystemResource} from '../../constants/resources';
+import { StorageInstanceKind, EventKind } from '../../types';
+
+const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
+
+const RecentEvent = (props) =>{
+  const [data, loaded, loadError] = useK8sWatchResource<StorageInstanceKind>(GetFlashSystemResource(props?.match?.params?.name, props?.match?.params?.namespace));
+  const namespace= loaded && !loadError? data?.[0]?.metadata.namespace: '';
+  const odfEventNamespaceKindFilter = (event: EventKind): boolean => {
+    const eventKind = event?.involvedObject?.kind;
+    const eventNamespace = getNamespace(event);
+    return eventNamespace === namespace || eventKind === PersistentVolumeClaimModel.kind;
+  };
+  const [events, eventsLoaded] = useK8sWatchResource(eventsResource);
+  return (
+    <RecentEventsBody
+      events={{ data: events, loaded: eventsLoaded } as any}
+      filter={odfEventNamespaceKindFilter}
+    />
+  );
+};
+
+export const ActivityCard: React.FC<any> = (props) => {
```

Comment:
Just a suggestion for better typings

```suggestion
import { match } from 'react-router';
export const ActivityCard: React.FC<{match: match<{namespace: string, name: string;}>}> = (props) => {
```

Reply:
it will report error
`TS2741: Property 'match' is missing in type '{ children?: ReactNode; }' but required in type '{ match: match<{ namespace: string; name: string; }>; }'`

Reply:
Ack. the [same](https://github.com/IBM/ibm-storage-odf-console/pull/10#discussion_r668654291) applies here